core/tests/test_utils.py

```diff
@@ -846,12 +1005,13 @@ class FailingFunction(FunctionWrapper):
     def __init__(self, f, exception, num_tries_before_success):
         """Create a new Failing function.
 
-        args:
-          - f: see FunctionWrapper.
-          - exception: the exception to be raised.
-          - num_tries_before_success: the number of times to raise an
-            exception, before a call succeeds. If this is 0, all calls will
-            succeed, if it is FailingFunction.INFINITY, all calls will fail.
+        Args:
+            f: func. See FunctionWrapper.
+            exception: str. The exception to be raised.
```

Comment:
>> Are you sure this is a str? What happens when you do type(self._exception)?

> It should be str.

This doesn't answer my question. Please explain the reasoning for why its type is str.

Reply:
How can I write type for exception? If it is the base Exception class, then it should be 'exceptions.Exception' . Or it can be any derived Exception class instead.

Reply:
```
Args:
    exception: Exception. The exception to be raised.
```

(i.e. use the least general class name that still covers all the cases; this also includes derived classes which are basically subtypes.)

Reply:
Its type is Exception as this is the method that can raise any exception inside this base Exception class.